python/python/bystro/prs/preprocess_for_prs.py

```diff
@@ -73,6 +71,34 @@ def _load_genetic_maps_from_feather(map_directory_path: str) -> dict[str, pd.Dat
     return genetic_maps
 
 
+def _extract_nomiss_dosage_loci(dosage_matrix_path: str) -> pd.DataFrame:
+    """
+    Reads a dataset from the provided dosage matrix Feather file path and filters out
+    rows where any column other than 'locus' contains a missing dosage value (represented by -1).
+
+    Args:
+    ----
+    dosage_matrix_path (str): The path to the Feather file containing the dataset.
+
+    Returns:
+    -------
+    pd.DataFrame: A DataFrame with a single 'locus' column containing loci with no missing
+    dosage values across all other columns.
+    """
+    dataset = ds.dataset(dosage_matrix_path, format="feather")
+    conditions = []
+    for column in dataset.schema.names:
+        if column != "locus":
+            condition = ds.field(column) != -1
+            conditions.append(condition)
+    combined_condition = conditions[0]
```

Comment:
This make sense as a first step; could @austinTalbot7241993 and you work together on an imputation strategy (maybe SoftImputeCV?) for future iterations?

Reply:
Maybe it makes sense to do imputation on the dosage matrix upstream of analyses overall rather than per analysis since we're imputing for ancestry as well? Would it make sense to have a version of the dosage matrix with imputation even as the dosage matrix is generated?